rr_viz/src/ross/slam/slam_supervisor.py

```diff
@@ -0,0 +1,376 @@
+            temp_timer = rospy.Timer(
+                rospy.Duration(.1), lambda _: self.waitTillDead_and_execute(self.run_localization), oneshot=True)
+        else:
+            print("Failed calling slam_killnodes_srv")
+
+    def run_localization(self):
+        _str = StringRequest()
+        _str.str = self.map_list_widget.currentItem().text().split(".")[0].strip()
+        trig_resp = self.launch_localization_srv.call(_str)
+
+        if trig_resp.success:
+            print(trig_resp.message)
+            self.mode_label.setText("Localization")
+            self.loaded_map_name = _str.str
+            self.loaded_map_label.setText(self.loaded_map_name)
+        else:
+            print("Failed calling slam_launch_localization_srv")
+        
+    def waitTillDead_and_execute(self, func):
+        if len(self.active_nodes) == 0:
+            func()
+        else:
+            rospy.Timer(
+                rospy.Duration(1), lambda _: self.waitTillDead_and_execute(func), oneshot=True)
+
+    def delete_map(self):
+        _str = StringRequest()
+        _str.str = self.map_list_widget.currentItem().text().split(".")[0].strip()
+
+        try:
+            trig_resp = self.list_maps_srv.call(TriggerRequest())
+        except Exception as e:
+            rospy.logwarn_throttle(
+                10, "failed to fetch maps: {}".format(e))
+            return
+
+        current_item = self.map_list_widget.currentItem()
+        self.localization_button.setEnabled(current_item is not None)
+
+        if trig_resp.success:
+            remote_maps = str(trig_resp.message).split(",")
+            if (_str.str == 'default_map'):
+                rospy.logwarn_throttle(10, "The default map cannot be deleted")
+                self.msg_to_show= "The default_map cannot be deleted."
+                self.message_popup()
+            elif (_str.str == self.loaded_map_name):
+                rospy.logwarn_throttle(10, "Map cannot be deleted")
+                self.msg_to_show = "'" + self.loaded_map_name + "' cannot be deleted while loaded."
+                self.message_popup()
+            else:
+                trig_resp = self.delete_map_srv.call(_str)
+                if trig_resp.success:
+                    print(trig_resp.message)
+                else:
+                    print("Failed calling slam delete map service")
+                    print(trig_resp.message)
+
+    def save_map(self):
+        _str = StringRequest()
+        map_name, ok = QInputDialog.getText(self, "Map file name", "Specify file name to save the map:")
+
+        if ok:
+            if map_name != '':
+                _str.str = string.replace(map_name, '.', '_')
+            else:
+                if not self.default_map_name:
+                    _str.str = randomTimeString()
+                else:
+                    _str.str = self.default_map_name
+
+            trig_resp = self.save_map_srv.call(_str)
+
+            if trig_resp.success:
```

Comment:
both cases here print the message - you can short this to a single if statement + always print the trig_resp.message